src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -11077,6 +11077,7 @@ void Assembler::evpbroadcastq(XMMRegister dst, Register src, int vector_len) {
 
 void Assembler::vpgatherdd(XMMRegister dst, Address src, XMMRegister mask, int vector_len) {
   assert(VM_Version::supports_avx2(), "");
+  assert(!needs_eevex(src.base(), src.index()), "does not support extended gprs as BASE or INDEX of address operand");
```

Comment:
Why this new assert in vpgatherdd?

Reply:
These are not promotable to extended EVEX encoding, BASE and INDEX registers of address operand must not be EGPRs. APX support is anyways enabled for AVX512 targets currently, but it's good to add safety assertions for completeness.

Reply:
But the index here is xmm register which could be xmm16 to xmm31 so the assert needs to be corrected. Also good to have the similar assert in vpgatherdq.

Reply:
That's correct, GATHER uses VSIB encoding where only base address is passed though a GPR while gather indexes are passed using vector register.